lib/doctor.ts

```diff
@@ -28,150 +28,207 @@ export class Doctor implements NativeScriptDoctor.IDoctor {
-			});
-		}
-
-		if (!sysInfoData.isAndroidSdkConfiguredCorrectly) {
-			result.push({
-				warning: "WARNING: The Android SDK is not installed or is not configured properly.",
-				additionalInformation: "You will not be able to run your apps in the native emulator. To be able to run apps" + EOL
-				+ "in the native Android emulator, verify that you have installed the latest Android SDK " + EOL
-				+ "and its dependencies as described in http://developer.android.com/sdk/index.html#Requirements" + EOL,
+			}), [
+				this.processSysInfoItem({
+					item: sysInfoData.adbVer,
+					infoMessage: "Your adb from the Android SDK is correctly installed.",
+					warningMessage: "adb from the Android SDK is not installed or is not configured properly. ",
+					additionalInformation: "For Android-related operations, the NativeScript CLI will use a built-in version of adb." + EOL
+						+ "To avoid possible issues with the native Android emulator, Genymotion or connected" + EOL
+						+ "Android devices, verify that you have installed the latest Android SDK and" + EOL
+						+ "its dependencies as described in http://developer.android.com/sdk/index.html#Requirements",
+					platforms: [Constants.ANDROID_PLATFORM_NAME]
+				}),
+				this.processSysInfoItem({
+					item: sysInfoData.isAndroidSdkConfiguredCorrectly,
+					infoMessage: "The Android SDK is installed.",
+					warningMessage: "The Android SDK is not installed or is not configured properly.",
+					additionalInformation: "You will not be able to run your apps in the native emulator. To be able to run apps" + EOL
+						+ "in the native Android emulator, verify that you have installed the latest Android SDK " + EOL
+						+ "and its dependencies as described in http://developer.android.com/sdk/index.html#Requirements",
+					platforms: [Constants.ANDROID_PLATFORM_NAME]
+				})
+			],
+			this.processValidationErrors({
+				warnings: this.androidToolsInfo.validateInfo(),
+				infoMessage: "A compatible Android SDK for compilation is found.",
 				platforms: [Constants.ANDROID_PLATFORM_NAME]
-			});
-		}
-
-		const androidToolsInfoValidationErrors = this.androidToolsInfo.validateInfo();
-		if (androidToolsInfoValidationErrors.length > 0) {
-			result = result.concat(androidToolsInfoValidationErrors);
-		}
-
-		const javacValidationErrors = this.androidToolsInfo.validateJavacVersion(sysInfoData.javacVersion);
-		if (javacValidationErrors.length > 0) {
-			result = result.concat(javacValidationErrors);
-		}
+			}),
+			this.processValidationErrors({
+				warnings: this.androidToolsInfo.validateJavacVersion(sysInfoData.javacVersion),
+				infoMessage: "Javac is installed and is configured properly.",
+				platforms: [Constants.ANDROID_PLATFORM_NAME]
+			})
+		);
 
 		if (this.hostInfo.isDarwin) {
-			if (!sysInfoData.xcodeVer) {
-				result.push({
-					warning: "WARNING: Xcode is not installed or is not configured properly.",
+			result = result.concat([
```

Comment:
Same comment about the brackets - you can skip creating an array here and lose the `[]` brackets